content/client-lib-development-guide/features.textile

```diff
@@ -378,14 +383,19 @@ h3(#realtime-channel). Channel
 * @(RTL11)@ If a channel enters the @DETACHED@ or @FAILED@ state, then all messages that are still queued for send on that channel should be deleted from the queue triggering a failure for the publish or presence methods invoked for those messages
 * @(RTL4)@ @Channel#attach@ function:
 ** @(RTL4a)@ If already @ATTACHED@ or @ATTACHING@ nothing is done
+** @(RTL4e)@ If the channel is in the @DETACHING@ state, it will result in an error, as the channel must complete the detach request before reattaching
```

Comment:
> I think we should also change RTN11b. Does that sound right?

Yes. From the point of view of the realtime system this is an even simpler case, because the closing connection and the new connection are completely independent. So the only issue is in managing the state of the client - you can `connect()` while the outgoing connection is `CLOSING`, but when you later receive the `CLOSED` you have to make sure that you don't indicate the new connection as `CLOSED` - the library has to check whether the `CLOSED` message applies to the new or the old connection.